src/webgpu/util/f32_interval.ts

```diff
@@ -169,15 +169,15 @@ export type F32Vector =
   | [F32Interval, F32Interval, F32Interval, F32Interval];
 
 /** Coerce F32Interval[] to F32Vector if possible */
-function isF32Vector(v: number[] | IntervalBounds[] | F32Interval[] | F32Vector): v is F32Vector {
```

Comment:
I'm not sure if this worked as intended before.

`F32Vector` is a `2`, `3` or `4` element array of `F32Interval`.

`F32Interval[] | F32Vector` is essentially the same as just saying `F32Interval[]`, as the former is a superset of the latter.

So `isF32Vector` would return `true` when passing a `2`, `3` or `4` sized-array of `F32Interval`, even if the intention was not to treat it as a vector - is that intentional?

If not, I think you need to make `F32Vector` a true class to treat it as a distinct type. This might not be a bad idea as the number of nested `[` `]` in these tests are becoming rather hard to mentally map to their semantic meaning. A `vec(1, 2)` style helper would make it obvious that you are dealing with vectors, and would help with the RTTI problems.

If so, then I'd suggest just dropping the the `| F32Vector` part of the parameter - it's not doing anything, and is adding more confusion to the helper.

The changes to this method don't appear to address this issue.

Reply:
True being returned for F32Interval[] with size 2, 3, 4 was intentional. This specifically get used when using .maps. The return type of .map on a F32Vector is a an array, not a F32Vector, even through for all intents and purposes it will behave like a F32Vector. It get used like this `toF32Vector(vec.map(foo))` when running an operation component-wise over a vector.

I have been somewhat resistant to introducing a F32Vector class, since that would lose a lot of the 'free' array functionality, unless we reimplemented it. I have considered it previously, and will take another look at it while implementing the first pass of unittests.

Reply:
I agree that the `| F32Vector` should be droppable